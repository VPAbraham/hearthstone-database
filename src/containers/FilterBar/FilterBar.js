import React, { Component } from 'react';
import './FilterBar.scss';
import PropTypes from 'prop-types'
import images from '../../assets/images';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setCardCollection, setFilterType, setFilterCriteria, toggleLoading } from '../../actions/index';
import { getCards } from '../../apiCalls/apiCalls';

export class FilterBar extends Component {

  async componentDidMount() {
    this.props.toggleLoading(true);
    let cards = await getCards();
    this.props.setCardCollection(cards);
    this.props.toggleLoading(false)
  }

  handleChange = (e) => {
    e.preventDefault()
    const filterType = e.currentTarget.className
    const filterCriteria = e.currentTarget.id
    this.props.setFilterCriteria(filterCriteria)
    this.props.setFilterType(filterType)
  }

  filterClickHandler = async (e) => {
    e.preventDefault();
    await this.handleChange(e);
    const { filterType, filterCriteria } = this.props;
    this.props.toggleLoading(true);
    const newCards = await getCards(filterType, filterCriteria)
    this.props.setCardCollection(newCards)
    this.props.toggleLoading(false)
  }


  render() {
    const {
      common, rare, uncommon, rareholo
    } = images;

    const pokemonTypes = ['bug', 'dark', 'dragon', 'electric', 
    'fairy', 'fighting','fire', 'flying', 'ghost', 'grass', 'ground', 
    'ice', 'normal', 'poison', 'psychic', 'rock', 'steel', 'water']

    const typeDisplay = pokemonTypes.map(type => {
      return (
        <li key={type}>
          <img className='types' id={type} src={images[type]} alt='type symbol' onClick={((e) => this.filterClickHandler(e))}/>
        </li>
      )
    })

    const rarityDisplay = ['common', 'uncommon', 'rare', 'rareholo'].map(rarity => {
      const imgSrc = eval(rarity)
      return (
        <li key={rarity}>
          <img className='rarity' id={rarity} src={imgSrc} alt='rarity gem' onClick={((e) => this.filterClickHandler(e))}/>
        </li>
      )
    })

    const pokemonSets = ['base1', 'base2', 'base3', 'base4', 'base5', 'gym1', 'gym2']

    const setDisplay = pokemonSets.map(set => {
      const imgSrc = ('https://images.pokemontcg.io/' + set + '/symbol.png')
      return (
        <li className='set' id={set} key={set} onClick={((e) => this.filterClickHandler(e))}>
          <p>{set}</p>
          <img  src={imgSrc}  alt='set icon'/>
        </li>
      )
    })

    return (
      <div className='filter-bar'>
        <button onClick={((e) => this.filterClickHandler(e))}>CLEAR FILTERS</button >
        <section className='mana-filter'>
          <h3>Type</h3>
          <div className='filter-divider'></div>
          <ul className='mana-ul-1'>
           {typeDisplay}
          </ul>
        </section>
        <section className='rarity-filter'>
            <h3>Rarity</h3>
            <div className='filter-divider'></div>
          <ul>
            {rarityDisplay}
          </ul>
        </section>
        <section className='set-filter'>
          <h3>Set</h3>
          <div className='filter-divider'></div>
          <ul>
            {setDisplay}
          </ul>
        </section>
      </div>
    )
  }
}


export const mapStateToProps = ({ filterCriteria, filterType }) => ({
  filterCriteria,
  filterType
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setCardCollection,
    setFilterCriteria,
    setFilterType,
    toggleLoading
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(FilterBar);

FilterBar.propTypes = {
  setCardCollection: PropTypes.func.isRequired,
  setFilterCriteria: PropTypes.func.isRequired,
  setFilterType: PropTypes.func.isRequired,
  toggleLoading: PropTypes.func.isRequired,
  filterCriteria: PropTypes.string.isRequired,
  filterType: PropTypes.string.isRequired
}