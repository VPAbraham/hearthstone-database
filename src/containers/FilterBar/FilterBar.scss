.filter-bar {
  width: 20%;
  height: 90vh;
  // background: gray;
  border-right: solid 1px gray;
  font-family: 'Quicksand', sans-serif;
  padding: 1.5rem 1rem;
  display: inline-block;

  h3 {
    font-size: 1.2rem;
  }
}

.filter-divider {
  width: 100%;
  height: 2px;
  background: #3700B3;
}

.mana-filter {
  display: flex;
  flex-direction: column;
  justify-content: space-between;


  .mana-ul-1,
  .mana-ul-2 {
    display: flex;
    list-style: none;
    justify-content: space-around;
    margin-top: 1.2rem;

    li {
      background: rgb(177, 175, 91);
      width: 20px;
    }
  }

}

.rarity-filter {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: 4rem;

  ul {
    display: flex;
    list-style: none;
    margin-top: 1.2rem;
    justify-content: space-evenly;

    li {

      img {
        height: 40px;
        width: auto;
      }
    }
  }
}

.set-filter {

  ul {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: 100%;

    li {
      display: flex;
      margin: 5px 0;
      justify-content: space-around;
      transition: 0.5s;
      border-radius: 5px;
      align-items: center;

      &:hover {
        background: rgb(65, 65, 65);
        transition: 0.5s;
      }


      p {
        text-align: left;
        width: 100%;
      }

      img {
        border-radius: 3px;
        background: #A1B1FD;
        height: 2rem;
      }
    }

  }
}