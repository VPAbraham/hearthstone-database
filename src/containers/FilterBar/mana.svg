<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="194px" height="169px" viewBox="0 0 194 169" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 3.7.2 (28276) - http://www.bohemiancoding.com/sketch -->
    <title>cristal</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <polygon id="path-1" points="115 27.5115057 194.674337 73.5115057 194.674337 165.511506 115 211.511506 35.3256629 165.511506 35.3256629 73.5115057"></polygon>
        <mask id="mask-2" maskContentUnits="userSpaceOnUse" maskUnits="objectBoundingBox" x="-4" y="-4" width="167.348674" height="192">
            <rect x="31.3256629" y="23.5115057" width="167.348674" height="192" fill="white"></rect>
            <use xlink:href="#path-1" fill="black"></use>
        </mask>
        <linearGradient x1="21.631282%" y1="120.811326%" x2="50%" y2="50%" id="linearGradient-3">
            <stop stop-color="#3976BB" offset="0%"></stop>
            <stop stop-color="#1A3459" offset="100%"></stop>
        </linearGradient>
        <linearGradient x1="43.6630636%" y1="-3.51757708%" x2="67.4109642%" y2="92.2646725%" id="linearGradient-4">
            <stop stop-color="#0C2242" offset="0%"></stop>
            <stop stop-color="#303F54" offset="100%"></stop>
        </linearGradient>
        <radialGradient cx="50%" cy="8.25352041%" fx="50%" fy="8.25352041%" r="105.879498%" id="radialGradient-5">
            <stop stop-color="#D4DFEA" offset="0%"></stop>
            <stop stop-color="#A3AFBB" offset="42.6857462%"></stop>
            <stop stop-color="#636F7C" offset="100%"></stop>
        </radialGradient>
        <linearGradient x1="0%" y1="0%" x2="50%" y2="100%" id="linearGradient-6">
            <stop stop-color="#657C9F" offset="0%"></stop>
            <stop stop-color="#354666" offset="100%"></stop>
        </linearGradient>
        <linearGradient x1="92.504131%" y1="100%" x2="39.1562682%" y2="0%" id="linearGradient-7">
            <stop stop-color="#72ADDF" offset="0%"></stop>
            <stop stop-color="#1351A3" offset="100%"></stop>
        </linearGradient>
        <linearGradient x1="-3.88844824%" y1="83.6471915%" x2="88.414519%" y2="83.6471915%" id="linearGradient-8">
            <stop stop-color="#1B3459" offset="0%"></stop>
            <stop stop-color="#2E5D96" offset="0%"></stop>
            <stop stop-color="#2C3C52" offset="100%"></stop>
        </linearGradient>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" opacity="0.732101213">
        <g id="Desktop-HD" transform="translate(-100.000000, -89.000000)">
            <g id="cristal" transform="translate(82.000000, 54.000000)">
                <g id="Polygon-1" transform="translate(115.000000, 119.511506) rotate(-330.000000) translate(-115.000000, -119.511506) ">
                    <use fill="#1E2129" fill-rule="evenodd" xlink:href="#path-1"></use>
                    <use stroke="#000000" mask="url(#mask-2)" stroke-width="8" xlink:href="#path-1"></use>
                </g>
                <polygon id="Path-1" fill="url(#linearGradient-3)" points="70.6640625 198.980469 116.236853 118.012864 161.371094 199.023438"></polygon>
                <polygon id="Path-1" fill="url(#linearGradient-4)" transform="translate(116.017578, 79.518151) scale(-1, -1) translate(-116.017578, -79.518151) " points="70.6640625 119.980469 116.236853 39.0128637 161.371094 120.023438"></polygon>
                <polygon id="Path-1" fill="url(#radialGradient-5)" transform="translate(149.767578, 99.085138) scale(-1, -1) rotate(60.000000) translate(-149.767578, -99.085138) " points="104.414062 138.681431 148.986853 59.4458764 195.121094 138.724399"></polygon>
                <polygon id="Path-1" fill="url(#linearGradient-6)" transform="translate(149.767578, 139.085138) scale(-1, 1) rotate(60.000000) translate(-149.767578, -139.085138) " points="104.414063 178.681431 148.986853 99.4458764 195.121094 178.724399"></polygon>
                <polygon id="Path-1" fill="url(#linearGradient-7)" transform="translate(80.767578, 140.085138) rotate(60.000000) translate(-80.767578, -140.085138) " points="35.4140625 179.681431 79.9868528 100.445876 126.121094 179.724399"></polygon>
                <polygon id="Path-1" fill="url(#linearGradient-8)" transform="translate(81.767578, 99.085138) scale(1, -1) rotate(60.000000) translate(-81.767578, -99.085138) " points="36.4140625 138.681431 80.9868528 59.4458764 127.121094 138.724399"></polygon>
                <polygon id="Path-7" fill="#FFFFFF" opacity="0.304920709" points="139.664062 104 155.683594 54.453125 118.625 117.296875 196.199219 117.296875"></polygon>
                <polygon id="Path-7" fill="#47A0FF" opacity="0.304920709" transform="translate(71.327359, 153.914621) rotate(179.000000) translate(-71.327359, -153.914621) " points="48.1632969 176.039621 69.5988438 122.492746 32.54025 185.336496 110.114469 185.336496"></polygon>
                <polygon id="Path-7" fill="#02070B" opacity="0.247084888" transform="translate(106.605697, 56.301560) rotate(300.000000) translate(-106.605697, -56.301560) " points="83.4416342 78.4265596 104.877181 24.8796846 67.8185873 87.7234346 145.392806 87.7234346"></polygon>
                <polygon id="Path-7" fill="#47A0FF" opacity="0.304920709" transform="translate(95.905421, 166.626952) rotate(356.000000) translate(-95.905421, -166.626952) " points="96.5680245 183.454398 110.296756 133.750029 73.5066876 196.593779 118.304154 199.503875"></polygon>
                <polygon id="Path-7" fill="#000000" opacity="0.181553172" transform="translate(84.532476, 83.638599) scale(-1, 1) rotate(2.000000) translate(-84.532476, -83.638599) " points="85.1950802 100.466045 98.9238114 50.7616768 62.1337433 113.605427 106.93121 116.515522"></polygon>
                <polygon id="Path-7" fill="#242E3C" opacity="0.27891791" transform="translate(145.544168, 154.211688) rotate(298.000000) translate(-145.544168, -154.211688) " points="146.206772 171.039134 159.935503 121.334765 123.145435 184.178515 167.942901 187.08861"></polygon>
            </g>
        </g>
    </g>
</svg>