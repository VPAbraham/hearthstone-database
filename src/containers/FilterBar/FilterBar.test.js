import React from 'react';
import { shallow } from 'enzyme';
import { FilterBar, mapStateToProps, matchDispatchToProps } from './FilterBar';

describe('FilterBar', () => {
  beforeEach(() => {
    const mockSetCardCollection = jest.fn()
    const mockSetFilterCriteria = jest.fn()
    const mocksetFilterType = jest.fn()
  })
  it('should match the snapshot', () => {
    let wrapper = shallow(
      <FilterBar 
        filterCriteria={''}
        filterType={''}
        setCardCollection={jest.fn()}
        setFilterType={jest.fn}
        setFilterCriteria={jest.fn}
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})