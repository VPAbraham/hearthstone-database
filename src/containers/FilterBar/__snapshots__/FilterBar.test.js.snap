// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should match the snapshot 1`] = `
<div
  className="filter-bar"
>
  <button
    onClick={[Function]}
  >
    CLEAR FILTERS
  </button>
  <section
    className="mana-filter"
  >
    <h3>
      Type
    </h3>
    <div
      className="filter-divider"
    />
    <ul
      className="mana-ul-1"
    >
      <li
        key="bug"
      >
        <img
          alt="type symbol"
          className="types"
          id="bug"
          onClick={[Function]}
          src="Bug.png"
        />
      </li>
      <li
        key="dark"
      >
        <img
          alt="type symbol"
          className="types"
          id="dark"
          onClick={[Function]}
          src="Dark.png"
        />
      </li>
      <li
        key="dragon"
      >
        <img
          alt="type symbol"
          className="types"
          id="dragon"
          onClick={[Function]}
          src="Dragon.png"
        />
      </li>
      <li
        key="electric"
      >
        <img
          alt="type symbol"
          className="types"
          id="electric"
          onClick={[Function]}
          src="Electric.png"
        />
      </li>
      <li
        key="fairy"
      >
        <img
          alt="type symbol"
          className="types"
          id="fairy"
          onClick={[Function]}
          src="Fairy.png"
        />
      </li>
      <li
        key="fighting"
      >
        <img
          alt="type symbol"
          className="types"
          id="fighting"
          onClick={[Function]}
          src="Fighting.png"
        />
      </li>
      <li
        key="fire"
      >
        <img
          alt="type symbol"
          className="types"
          id="fire"
          onClick={[Function]}
          src="Fire.png"
        />
      </li>
      <li
        key="flying"
      >
        <img
          alt="type symbol"
          className="types"
          id="flying"
          onClick={[Function]}
          src="Flying.png"
        />
      </li>
      <li
        key="ghost"
      >
        <img
          alt="type symbol"
          className="types"
          id="ghost"
          onClick={[Function]}
          src="Ghost.png"
        />
      </li>
      <li
        key="grass"
      >
        <img
          alt="type symbol"
          className="types"
          id="grass"
          onClick={[Function]}
          src="Grass.png"
        />
      </li>
      <li
        key="ground"
      >
        <img
          alt="type symbol"
          className="types"
          id="ground"
          onClick={[Function]}
          src="Ground.png"
        />
      </li>
      <li
        key="ice"
      >
        <img
          alt="type symbol"
          className="types"
          id="ice"
          onClick={[Function]}
          src="Ice.png"
        />
      </li>
      <li
        key="normal"
      >
        <img
          alt="type symbol"
          className="types"
          id="normal"
          onClick={[Function]}
          src="Normal.png"
        />
      </li>
      <li
        key="poison"
      >
        <img
          alt="type symbol"
          className="types"
          id="poison"
          onClick={[Function]}
          src="Poison.png"
        />
      </li>
      <li
        key="psychic"
      >
        <img
          alt="type symbol"
          className="types"
          id="psychic"
          onClick={[Function]}
          src="Psychic.png"
        />
      </li>
      <li
        key="rock"
      >
        <img
          alt="type symbol"
          className="types"
          id="rock"
          onClick={[Function]}
          src="Rock.png"
        />
      </li>
      <li
        key="steel"
      >
        <img
          alt="type symbol"
          className="types"
          id="steel"
          onClick={[Function]}
          src="Steel.png"
        />
      </li>
      <li
        key="water"
      >
        <img
          alt="type symbol"
          className="types"
          id="water"
          onClick={[Function]}
          src="Water.png"
        />
      </li>
    </ul>
  </section>
  <section
    className="rarity-filter"
  >
    <h3>
      Rarity
    </h3>
    <div
      className="filter-divider"
    />
    <ul>
      <li
        key="common"
      >
        <img
          alt="rarity gem"
          className="rarity"
          id="common"
          onClick={[Function]}
          src="common.svg"
        />
      </li>
      <li
        key="uncommon"
      >
        <img
          alt="rarity gem"
          className="rarity"
          id="uncommon"
          onClick={[Function]}
          src="rare.svg"
        />
      </li>
      <li
        key="rare"
      >
        <img
          alt="rarity gem"
          className="rarity"
          id="rare"
          onClick={[Function]}
          src="epic.svg"
        />
      </li>
      <li
        key="rareholo"
      >
        <img
          alt="rarity gem"
          className="rarity"
          id="rareholo"
          onClick={[Function]}
          src="legendary.svg"
        />
      </li>
    </ul>
  </section>
  <section
    className="set-filter"
  >
    <h3>
      Set
    </h3>
    <div
      className="filter-divider"
    />
    <ul>
      <li
        className="set"
        id="base1"
        key="base1"
        onClick={[Function]}
      >
        <p>
          base1
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/base1/symbol.png"
        />
      </li>
      <li
        className="set"
        id="base2"
        key="base2"
        onClick={[Function]}
      >
        <p>
          base2
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/base2/symbol.png"
        />
      </li>
      <li
        className="set"
        id="base3"
        key="base3"
        onClick={[Function]}
      >
        <p>
          base3
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/base3/symbol.png"
        />
      </li>
      <li
        className="set"
        id="base4"
        key="base4"
        onClick={[Function]}
      >
        <p>
          base4
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/base4/symbol.png"
        />
      </li>
      <li
        className="set"
        id="base5"
        key="base5"
        onClick={[Function]}
      >
        <p>
          base5
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/base5/symbol.png"
        />
      </li>
      <li
        className="set"
        id="gym1"
        key="gym1"
        onClick={[Function]}
      >
        <p>
          gym1
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/gym1/symbol.png"
        />
      </li>
      <li
        className="set"
        id="gym2"
        key="gym2"
        onClick={[Function]}
      >
        <p>
          gym2
        </p>
        <img
          alt="set icon"
          src="https://images.pokemontcg.io/gym2/symbol.png"
        />
      </li>
    </ul>
  </section>
</div>
`;
